notebooks/117-optimize-preprocessing/117-optimize-preprocessing.ipynb

```diff
@@ -0,0 +1,498 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Perform inference "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "output_layer = compiled_model.output(0)\n",
+    "\n",
+    "result = compiled_model(input_tensor)[output_layer]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compare results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compare results on one image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def results(input_tensor, compiled_model):\n",
+    "    output_layer = compiled_model.output(0)\n",
+    "\n",
+    "    results = compiled_model(input_tensor)[output_layer][0]\n",
+    "\n",
+    "    top_indices = np.argsort(results)[-5:][::-1]\n",
+    "    top_softmax = results[top_indices]\n",
+    "\n",
+    "    return top_indices, top_softmax\n",
+    "\n",
+    "# Convert the inference result to a class name.\n",
+    "imagenet_classes = open(\"../data/datasets/imagenet/imagenet_2012.txt\").read().splitlines()\n",
+    "imagenet_classes = ['background'] + imagenet_classes\n",
+    "\n",
+    "# get result for inference with preprocessing api\n",
+    "top_indices, top_softmax = results(ppp_input_tensor, compiled_model_with_preprocess)\n",
+    "\n",
+    "print(\"Result of inference with preprocessing api:\")\n",
+    "for index, softmax_probability in zip(top_indices, top_softmax):\n",
+    "    print(f\"{imagenet_classes[index]}, {softmax_probability:.5f}\")\n",
+    "\n",
+    "print(\"\\n\")\n",
+    "# get result for inference with the manual preparing of the image\n",
+    "top_indices, top_softmax = results(input_tensor, compiled_model)\n",
+    "\n",
+    "print(\"Result of inference with manual image setup:\")\n",
+    "for index, softmax_probability in zip(top_indices, top_softmax):\n",
+    "    print(f\"{imagenet_classes[index]}, {softmax_probability:.5f}\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compare performance"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# check performance for inference with preprocessing api\n",
+    "num_images = 1000\n",
+    "\n",
+    "start = time.perf_counter()\n",
+    "\n",
+    "for _ in range(num_images):\n",
+    "    compiled_model_with_preprocess(ppp_input_tensor)\n",
+    "\n",
+    "end = time.perf_counter()\n",
+    "time_ir = end - start\n",
+    "\n",
+    "print(\n",
+    "    f\"IR model in OpenVINO Runtime/CPU with preprocessing API: {time_ir/num_images:.4f} \"\n",
+    "    f\"seconds per image, FPS: {num_images/time_ir:.2f}\"\n",
+    ")\n",
+    "\n",
+    "# check performance for inference with the manual preparing of the image\n",
```

Comment:
Shall we add opencv manual processing into this loop, which is also included in execution graph with preprocessing API

Reply:
done